processor/filterprocessor/logs.go

```diff
@@ -18,21 +18,41 @@ import (
 	"context"
 	"fmt"
 
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterlog"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllog"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor/internal/common"
 )
 
 type filterLogProcessor struct {
 	cfg            *Config
 	excludeMatcher filterlog.Matcher
 	includeMatcher filterlog.Matcher
 	logger         *zap.Logger
+	logConditions  []*ottl.Statement[ottllog.TransformContext]
 }
 
 func newFilterLogsProcessor(logger *zap.Logger, cfg *Config) (*filterLogProcessor, error) {
+	if cfg.Logs.LogConditions != nil {
```

Comment:
Add a warning if using the old config? And mark the old one as deprecated?

If you want you can do a separate PR.

Reply:
I still need to check that OTTL has 100% of the features offered by the existing configuration options so I'd opt not to add any Deprecated comments/warnings yet.